app/src/main/java/ch/epfl/sdp/appart/ad/AdCreationViewModel.java

```diff
@@ -53,8 +53,9 @@ public AdCreationViewModel(DatabaseService db, LoginService ls) {
     public CompletableFuture<Boolean> confirmCreation() {
         User user = ls.getCurrentUser();
         Ad ad = new Ad(title, price, pricePeriod, street, city, user.getName(), user.getUserId(), description,
-                new ArrayList<>(), VRTourEnable);
-        CompletableFuture<String> result = db.putAd(ad, photosUri);
+                new ArrayList<>(), new ArrayList<>(), VRTourEnable);
+        CompletableFuture<String> result = db.putAd(ad, photosUri, panoramasUris);
+
```

Comment:
don't you have to manage the case of future failure?

Reply:
Isn't this the case bellow ?
```
public CompletableFuture<Boolean> confirmCreation() {
        User user = ls.getCurrentUser();
        Ad ad = new Ad(title, price, pricePeriod, street, city, user.getName(), user.getUserId(), description,
                new ArrayList<>(), new ArrayList<>(), VRTourEnable);
        CompletableFuture<String> result = db.putAd(ad, photosUri, panoramasUris);

        return result.thenApply(s -> {
            user.addAdId(s);
            return true;
        }).exceptionally(e -> false);

    }
```

Reply:
I'm sorry in the "File changes" I did't see the last line of code.

Reply:
No problems